src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts

```diff
@@ -48,6 +52,32 @@ export class WorkspaceIdConsumerWrapper {
     return type === UI_SETTINGS_SAVED_OBJECTS_TYPE;
   }
 
+  private validateObjectInAWorkspace<T>(
+    object: SavedObject<T>,
+    workspace: string,
+    request: OpenSearchDashboardsRequest
+  ) {
+    // Data source is a workspace level object, validate if the request has access to the data source within the requested workspace.
+    if (validateIsWorkspaceDataSourceAndConnectionObjectType(object.type)) {
+      if (!!getWorkspaceState(request).isDataSourceAdmin) {
+        return true;
+      }
+      // Deny access if the object is a global data source (no workspaces assigned)
+      if (!object.workspaces || object.workspaces.length === 0) {
```

Comment:
I am thinking there will be an issue that dashboard admin can list all the data sources but if they switch to a global data source, they will be able to denied. Can we allow dashboard admin to `get` data sources when outside workspace?

Reply:
If user is outside the workspace, this method will not be called and bulkGet/Get will return object directly.

Reply:
Oh, yes. Thanks for that.